src/algokit/cli/tasks/mint.py

```diff
@@ -0,0 +1,163 @@
+import logging
+import math
+from pathlib import Path
+
+import click
+
+from algokit.cli.tasks.utils import (
+    get_account_with_private_key,
+    get_asset_explorer_url,
+    get_transaction_explorer_url,
+    load_algod_client,
+)
+from algokit.core.tasks.ipfs import get_web3_storage_api_key
+from algokit.core.tasks.mint.mint import MetadataStandard, mint_token
+from algokit.core.tasks.mint.models import TokenMetadata
+
+logger = logging.getLogger(__name__)
+
+
+def _validate_inputs(total: int, decimals: int) -> None:
+    # Validate total and decimals
+    if not (total == 1 or (total % 10 == 0 and total != 0)):
+        raise click.ClickException("Total must be 1 or a power of 10 larger than 1 (10, 100, 1000, ...).")
+    if not ((total == 1 and decimals == 0) or (total != 1 and decimals == int(math.log10(total)))):
+        raise click.ClickException(
+            "Number of digits after the decimal point must be 0 for a pure NFT, or "
+            "equal to the logarithm in base 10 of total number of units for a fractional NFT."
+        )
+
+
+@click.command(
+    name="mint",
+    help="Mint a new fungible or non-fungible assets on Algorand.",
+)
+@click.option(
+    "--creator",
+    "-r",
+    required=True,
+    prompt="Please enter the address or alias of the asset creator",
+    help="Address or alias of the asset creator.",
+    type=click.STRING,
+)
+@click.option(
+    "--asset-name",
+    type=click.STRING,
+    required=True,
+    prompt="Please enter the asset name",
+    help="Asset name.",
+)
+@click.option(
+    "--unit-name",
+    type=click.STRING,
+    required=True,
+    prompt="Please enter the unit name",
+    help="Unit name of the asset.",
+)
+@click.option(
+    "--total",
+    type=click.INT,
+    required=True,
+    prompt=(
+        "Please enter the total supply. Set 1 for a pure NFT, or a power of 10 "
+        "larger than 1 (10, 100, 1000, ...) for a fractional NFT."
+    ),
+    help=(
+        "Total supply of the asset. Set 1 for a pure NFT, or a power of 10 "
+        "larger than 1 (10, 100, 1000, ...) for a fractional NFT."
+    ),
+)
+@click.option(
+    "--decimals",
+    type=click.INT,
+    required=False,
+    prompt="""Please enter the number of decimals. Set 0 for a pure NFT,
+        or equal to the logarithm in base 10 of total supply for a fractional NFT.""",
+    help="""Number of decimals. Set 0 for a pure NFT, ",
+        "or equal to the logarithm in base 10 of total supply for a fractional NFT.""",
+)
+@click.option(
+    "--image-path",
+    "image_path",
+    type=click.Path(exists=True, dir_okay=False, file_okay=True, resolve_path=True, path_type=Path),
+    prompt="Please enter the path to the ARC19 compliant asset metadata file to be uploaded to IPFS",
```

Comment:
I'm not sure the prompt and help text are correct? They talk about metadata rather than image.